bytecomp/translcore.ml

```diff
@@ -899,7 +904,7 @@ and transl_match e arg pat_expr_list partial =
     Lstaticcatch
       (Ltrywith (Lstaticraise (static_exception_id, body), id,
                  Matching.for_trywith (Lvar id) exn_cases),
-       (static_exception_id, val_ids),
+       (static_exception_id, (List.map (fun id -> id, Pgenval) val_ids)),
```

Comment:
The kind information is available for that case:
```OCaml
let f x =
  match x () with
  | exception Not_found -> 1.
  | n -> n
```
The catch should be annotated with float here
```
       (function x/5 : float
         (catch
           (try (exit 1 (apply x/5 0a)) with exn/9
             (if (== exn/9 (field 8 (global Stdlib!))) 1. (reraise exn/9)))
          with (1 n/6) n/6)))
```

Reply:
Thanks, done.